qiskit_nature/second_q/mappers/qubit_converter.py

```diff
@@ -167,6 +170,10 @@ def z2symmetries(self) -> Z2Symmetries:
         """
         return copy.deepcopy(self._z2symmetries)
 
+    def _check_reset_mapper(self) -> None:
```

Comment:
Maybe you could add some sort of explanation for a reader of this code?
I.e. that this method is used to preserve the original behavior of the `QubitConverter` class with respect to how it deals with `two_qubit_reduction` and `num_particles` being set, whereas the new `ParityMapper` only has a single argument (namely `num_particles`) to handle this.